### PR TITLE
solar sample

### PR DESCRIPTION
**Pull request by moonlanderr**
solar energy prediction 

-----

# Checklist

Please go through each entry in the below checklist and mark an 'X' if that condition has been met. Every entry should be marked with an 'X' to be get the Pull Request approved.

- [ ] All `import`s are in the first cell? First block of imports are standard libraries, second block are 3rd party libraries, third block are all `arcgis` imports? Note that in some cases, for samples, it is a good idea to keep the imports next to where they are used, particularly for uncommonly used features that we want to highlight.
- [ ] All `GIS` object instantiations are one of the following?
    - `gis = GIS()`
    - `gis = GIS('https://www.arcgis.com', 'arcgis_python', 'P@ssword123')`
    - `gis = GIS(profile="your_online_profile")`
    - `gis = GIS('https://pythonapi.playground.esri.com/portal', 'arcgis_python', 'amazing_arcgis_123')`
    - `gis = GIS(profile="your_enterprise_portal")`
- [ ] If this notebook requires setup or teardown, did you add the appropriate code to `./misc/setup.py` and/or `./misc/teardown.py`?
- [ ] If this notebook references any portal items that need to be staged on AGOL/Python API playground, did you coordinate with a Python API team member to stage the item the correct way with the api\_data\_owner user?
- [ ] Code refactored & split out across multiple cells, useful comments?
- [ ] Consistent voice/tense/narrative style? Thoroughly checked for typos?
- [ ] All images used like `` instead of ``? All map widgets contain a static image preview? (Call `mapview_inst.take_screenshot()` to do so)
- [ ] All file paths are constructed in an OS-agnostic fashion with `os.path.join()`? (Instead of `r"\foo\bar"`, `os.path.join(os.path.sep, "foo", "bar")`, etc.)
- [ ] **IF YOU WANT THIS SAMPLE TO BE DISPLAYED ON THE DEVELOPERS.ARCGIS.COM WEBSITE**, ping @ DavidJVitale so he can add it to the list for the next deploy

### CODE REVIEW

**Comment by priyankatuteja** [comment]
@guneetmutreja can you do the first round of review for this?

**Comment by priyankatuteja** [comment]
@moonlanderr there is an extra file in this commit, please remove that.

**Comment by priyankatuteja** [comment]
@SurajBaloni Could you also review this

**Comment by AtmaMani** [comment]
@moonlanderr can you notify @moonlanderr and I once you address Priyanka's comments? We want to target this for July 7th content freeze

**Comment by moonlanderr** [comment]
<p>df.head() added</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>corrected</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>all warnings now removed</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>Let it be separate, since these two are actually two separate notebooks on fcn and MLModel combined into one, and they can be run independently</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>Since this is not a model metric, It can be separate cell.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>Let it be separate, since these two are actually two separate notebooks on fcn and MLModel combined into one. So the user could either run the MLModel part or the fcn part independently.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>it can depend on the user according to the field names the user will have</p><p> </p><p>pd.to_datetime is retained , as explained above</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>since we have been using query().sdf throughout, let it be  there for consistency</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>this shows the place where the import is being used, for the user it points out where the imports are in action </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>this shows that actual code where the imports are being implemented which will be helpful for the user</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>the original explanation is retained since it describes the significance of the new MLModel and the flexibility it provides along with a brief for adding the parameters of the sklearn ml model</p><p> </p><p>Also here FullyConnectedNetwork  means fully connected not fully trained.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>This new paragraph removes lot of information about the variable and their meaning, and does not explain it fully, Lets retain the original explanation, which details shortwave energy received by the surface, the dayls_meaning day length and how they are affecting the dependent variable.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p><em>"The primary information consists of the daily generation of energy in kilowatt-hour(KWh) given here in the field name&nbsp;kWh_filled&nbsp;for each of the selected 10 solar photovoltaic power plants in the City of Calgary. The field&nbsp;capacity_f&nbsp;indicates the capacity factor which is obtained after normalizing the&nbsp;kWh_filled&nbsp;by the peak capacity of each solar photovoltaic sites, which will be used here as the dependent variable."</em></p><p> </p><p>The kWH_filled column is defined here in the previous cell, which has to be referred while understanding this explanation. Here it means that this is the y-variable, and the underlined line means that plot shows that the last plot of kWH_filled variable is similar with all the other plots representing the other variables , hence indicating a relationship. </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>Telus spark is an extreme case,  while hillhurst is the general trend, keeping in mind that distribution will always vary for different power plant. I have checked plots of the other stations and selected this as a general example.  </p><p> </p><p>This conversion code is setting it as datetime index from a datetime field and also renaming it, unless the datetime conversion is done even for datetime field it sometimes give error. Including the datetime conversion makes it more robust and ensures the conversion for all kind of datasets. Since we are not sure if the data used by the user will have the datetime format or not,  this step will ensure the user convert this to datetime if its not a datetime.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added field names in bold, </p><p> </p><p>filed names will vary with the dataset used by the user, so this line explains what the field names means. </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>added </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>removed</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>corrected</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>shifting means changing or usually used in case of relocating something, moving away is a phrase and is appropriate here. Renewable energy sources is self explanatory .</p><p> </p><p> Lets retain the rest of the introduction there are valuable information in it, which should not be removed, such as the hypothesis is the base of any study , which is essential to tell the reader why and how the  problem is formulated, and the various assumption it is based on.  Also there are parts where it will tell the user how this study will help in a real world scenario. Also it is necessary to build up a story and provide a brief summary of what is going to be there.</p><p> </p><p>Added the last paragraph with edits.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>corrected</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
@AtmaMani addressed Priyanka's comments, please check

**Comment by priyankatuteja** [comment]
<p>I think the the data you have provided in the form of feature layer is specific to this sample and users can even have a complete different set of explanatory variable names like date field, So, i would suggest to reduce unnecessary the lines of code .</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
<p>I have run this sample without filtering the dataframe by a specific site. It still generates the same plot, so we can avoid one extra line of code to optimize code.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
<p>Prerequisites section is still missing. (scikit learn and seaborn dependencies required)</p><p> </p><p>I feel the paragraphs that i had cut were more appropriate under the predict or model forecast section. 2nd paragraph looks repetitive of last two lines from 1st paragraph.</p><p> </p><p>Additionally, @BP-Ent could guide better.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
<p>Instead of providing unnecessary conversion of datetime column again to datetime, we could write a comment about the date field type required since users might have different formats of data which could need more lines of code to get the date field in this format.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
<p>I won't recommend repeating same 13 lines twice in the same notebook. Also as you want to keep this notebook generalizable for users, a function will give users the flexibility to add their custom variables.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
<p><s>Finally, performance between the two methods will be compared in terms of model training and validation accuracy.</s></p><p>We should not aim to compare our models but show utility of both of these. </p><p> </p><p><s>1) Fully Connected Network - First a deep learning framework called Fully Connected Network (fcn) available in the arcgis.learn module in ArcGIS API for Python is used.</s></p><p>This line has been mentioned as part of introduction, it is repetitive.</p><p> </p><p><s>2) Machine Learning Model - In the second option, one of the machine learning algorithms from scikit-learn will be implemented via the MLModel framework available in arcgis.learn. This framework can deploy any ML algorithm from the scikit-learn library just by passing the name of the algorithm and its relevant parameters as keyword arguments.</s></p><p>scikit learn also supports other ML algorithms including clustering (unsupervised) which is not supported through MLModel as per the current link provided in api reference (<a href="https://scikit-learn.org/stable/supervised_learning.html#supervised-learning" target="_blank">https://scikit-learn.org/stable/supervised_learning.html#supervised-learning</a>) which is why i suggested the following lines:</p><p> </p><p>The MLModel allows you to use any regression or classification model from scikit learn whereas FullyConnectedNetwork allows you to train a fully trained deep neural network with a specified number of layers.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
<p>@moonlander I have marked some changes unresolved, please check those again. There is an extra file in this PR which needs to be removed as well.</p>

**Comment by moonlanderr** [comment]
<p>I have added the scikit-learn dependencies now in the import section, not using seaborn anymore</p><p> </p><p>repetitive part now removed</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p><em>It still generates the same plot, - </em>not clear to me. A site must be mentioned to generate a plot, and each plot should be different, but can have a similar trend. This is an important visualization, should be there.</p><p> </p><p>The conversion is necessary as mentioned before. Its better to show the implementaion than describing about the conversion.</p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>comparing different model performances makes a study interesting especially where different methods are applied, </p><p> </p><p>The repetition is removed from the introduction, and this line is retained here.</p><p> </p><p>added </p><blockquote><em>any regression or classification model </em></blockquote><p> </p><p> </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>most of the metrics are now removed, it is now using only R-squared to keep it consistent with the training metric. Now has only 2 line of code</p><p> </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by moonlanderr** [comment]
<p>this is described in the final column.</p><p> </p>

---

 <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/688/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by priyankatuteja** [comment]
@BP-Ent Could you help in reviewing this sample?

**Comment by priyankatuteja** [comment]
https://github.com/Esri/arcgis-python-api/pull/782 duplicate

**Comment by priyankatuteja** [comment]
@BP-Ent Thanks for the review.
